### DISCUSSION TITLE
How can i make use of Parcel v2 to load a custom config file?

### DISCUSSION THREAD

**Discussion by hamzamihaidanielx** [parcel-bundler/parcel#5559] [Q&A] [answered]
Hello, Uhm, I'm trying to make the v2 parcel to load my custom config file, but the documentation over [https://v2.parceljs.org/features/parcel-api/](https://v2.parceljs.org/features/parcel-api/) it's pretty limited and does not tell you more...

Thank you kindly!

**Comment by mischnic** [comment, marked as answer]
What are you trying to achieve?

**Reply by hamzamihaidanielx** [reply]
Well basically, using a hook in order to merge some of my react (components)modules's package.json into the main package.json of the web app. And would love also to output the assets to an in-memory file system and even more.

**Reply by mischnic** [reply]
You can pass `targets` in the parameter of `new Parcel`. That has the same structure as `targets` in package.json: https://v2.parceljs.org/configuration/package-json/. Not sure if that covers the things you want to merge though.

**Reply by hamzamihaidanielx** [reply]
Nah not quite..., so there is no way to run a custom parcel config file based on the parcel-api link above?

**Reply by mischnic** [reply]
You mean `.parcelrc`? You can pass a filepath to `config` https://v2.parceljs.org/plugin-system/api/#InitialParcelOptions

**Reply by hamzamihaidanielx** [reply]
Then how can I even for instance, implement the second example as shown [https://v2.parceljs.org/features/parcel-api/](https://v2.parceljs.org/features/parcel-api/)? The documentation there sadly does not tell how to even let Parcel know to load that.

Thank you!

**Reply by mischnic** [reply]
Just replace `defaultConfig` with `config`. And instead of `require.resolve("@parcel/config-default")` you can pass a path to a different json file.

**Reply by hamzamihaidanielx** [reply]
Interesting, but that example over there, it's an es6 file so this does not work 😭

**Reply by mischnic** [reply]
Ah, this is Babel's ESM version. This should work:
```js
const path = require("path");
const {default: Parcel, createWorkerFarm } = require("@parcel/core");
const { NodeFS, MemoryFS } = require("@parcel/fs");
```

**Reply by hamzamihaidanielx** [reply]
Yep but this still does not tell me how to let Parcel know use this tho. Tried with filePath but won't work.

**Reply by mischnic** [reply]
filePath?

You'll have to put this into a JS file and run it with `node thatfile.js`

**Reply by hamzamihaidanielx** [reply]
<img width="804" alt="image" src="https://user-images.githubusercontent.com/12731515/103445135-20cd2080-4c79-11eb-99c3-d2a491ebe009.png">

Started with something basic, but regardless of what I do and try to configure as much as close to what's in [here](https://github.com/parcel-bundler/parcel/blob/e6a2a6312a472e70c353b3de9978b4f2a2b3406f/packages/core/parcel/src/cli.js#L185), I always get a white template with 404 - Page not found.

Any idea what may cause this? My thinking is that it has something to do with the public, but...

Thank you!

**Reply by hamzamihaidanielx** [reply]
Ahh, I am such an idiot, the entry should've been the .html not the .js. Seems to work just fine now! 😊

**Comment by kungfooman** [comment]
This works for me: `npx parcel build --config .parcelrc-rti test.rti.js`

But I can't figure out how to change the dist dir of that target now... neither config nor command line options seem to be respected.
